include/nbl/builtin/hlsl/prefix_sum_blur/box_sampler.hlsl

```diff
@@ -0,0 +1,183 @@
+#include "nbl/builtin/hlsl/cpp_compat.hlsl"
+#include "nbl/builtin/hlsl/enums.hlsl"
+#include "nbl/builtin/hlsl/macros.h"
+
+#ifndef _NBL_BUILTIN_BOX_SAMPLER_INCLUDED_
+#define _NBL_BUILTIN_BOX_SAMPLER_INCLUDED_
+
+namespace nbl
+{
+namespace hlsl
+{
+namespace prefix_sum_blur
+{
+
+template<typename PrefixSumAccessor, typename T>
+struct BoxSampler
+{
+    using prefix_sum_accessor_t = PrefixSumAccessor;
+
+    PrefixSumAccessor prefixSumAccessor;
+    uint16_t wrapMode;
+    uint16_t linearSize;
+    T normalizationFactor;
+
+    T operator()(float32_t ix, float32_t radius, float32_t borderColor)
+    {
+        const float32_t alpha = frac(radius);
+        const float32_t rightIdx = float32_t(ix) + radius;
+        const float32_t leftIdx = float32_t(ix) - radius - 1;
+        const int32_t lastIdx = linearSize - 1;
+        const int32_t rightFlIdx = (int32_t)floor(rightIdx);
+        const int32_t rightClIdx = (int32_t)ceil(rightIdx);
+        const int32_t leftFlIdx = (int32_t)floor(leftIdx);
+        const int32_t leftClIdx = (int32_t)ceil(leftIdx);
+
+        assert(linearSize > 1);
+
+        T result = 0;
+        if (rightClIdx < linearSize)
+        {
+            result += lerp(prefixSumAccessor.template get<T, uint32_t>(rightFlIdx), prefixSumAccessor.template get<T, uint32_t>(rightClIdx), alpha);
+        }
+        else
+        {
+            switch (wrapMode) {
+            case ETC_REPEAT:
+            {
+                const T last = prefixSumAccessor.template get<T, uint32_t>(lastIdx);
+                const T floored = prefixSumAccessor.template get<T, uint32_t>(rightFlIdx % linearSize) + last;
+                const T ceiled = prefixSumAccessor.template get<T, uint32_t>(rightClIdx % linearSize) + last;
+                result += lerp(floored, ceiled, alpha);

```

Comment:
this is surely broken, see https://github.com/Devsh-Graphics-Programming/Nabla/pull/819#discussion_r1923818897

your last isn't even multiplied by how many multiples of linearSize you're outside the range 🤦